core/services/ocrcommon/data_source_test.go

```diff
@@ -112,12 +114,13 @@ func Test_CachedInMemoryDataSourceErrHandling(t *testing.T) {
 		dsCache, err := ocrcommon.NewInMemoryDataSourceCache(ds, &mockKVStore, time.Hour*100)
 		require.NoError(t, err)
 		changeResultValue(runner, "-1", true, false)
+		require.NoError(t, dsCache.Start(testutils.Context(t)))
```

Comment:
The `Close()` at the end will never execute if you fail on an earlier `require`, so it is best to use something like `servicetest.Run` which handles scheduling the close for cleanup:
```suggestion
		servicetest.Run(t, dsCache)
```
```go
// Run fails tb if the service fails to start or close.
func Run[R Runnable](tb TestingT, r R) R {
	tb.Helper()
	require.NoError(tb, r.Start(tests.Context(tb)), "service failed to start")
	tb.Cleanup(func() { assert.NoError(tb, r.Close(), "error closing service") })
	return r
}
```

Reply:
cool, didn't know about servicetest